### DISCUSSION TITLE
Is there a way to test a specific function or module?

### DISCUSSION THREAD

**Discussion by prestontw** [gleam-lang/gleam#3626] [Questions & support] [answered]
I know I can run a specific binary with `gleam run -m <binary>`, but does something similar exist for testing? I have been changing the names of all of the other tests I don't want to run so they aren't picked up by `gleam test`. I am on `gleam 1.4.1`.

**Comment by lpil** [comment, marked as answer]
The "gleam test" command runs the main function in your APPNAME_test module, you can do anything you want in there. Gleam doesn't have an opinions about what your test code does.

**Reply by prestontw** [reply]
Great, thank you! This really helped my mental model of the default testing setup (through `gleam new`). Is this something that's already documented? I don't think I saw it in the language tour, but this is maybe too in the weeds for the tour. If it's helpful, I would be happy to add some documentation, but I would need some guidance on where to put it.

**Reply by lpil** [reply]
The tour covers the language rather than the build tool. This bit is explained in the "writing Gleam" tutorial which is linked to at the end of the tour.

https://gleam.run/writing-gleam/

**Comment by ToddG** [comment]
[glacier](https://hexdocs.pm/glacier/index.html) might be what the OP is looking for
